### ISSUE TITLE
How does PySCF calculate total energy?

### ISSUE THREAD

**Issue by lela2011** [pyscf/pyscf#1404]
I am writing my own program for restricted Hartree Fock theory. E.g. I'm calculating orbital energies and the coefficients as a result of the Roothaan Hall equation. I use PySCF to check whether my program is working properly, e.g. I compare the molecular energies. For total energies the program is not returning the same values though. I am therefore interested in how PySCF calculates the total energy of a molecule based on the matrices and eigen-energies that result from the scf iteration.

TLDR: What formula does PySCF use to calculate the total molecular energy for Hartree Fock?

**Comment by pwborthwick** [comment]
I had a quick look in your repo *maturaabbeit* which I'm assuming is the code you are referring to? From what I can see you are computing the energy as the sum of the orbital eigenvalues. The SCF energy  is not $\displaystyle \sum_{p} \epsilon_p$ or even $\displaystyle \sum_{i}$ over the occupied orbitals. The SCF ground state energy is $E_{0}^{el} = \displaystyle \sum_{pq} D_{pq} (F_{pq} + H^{core}_{pq})$ 

where $D_{pq} = \displaystyle \sum_{i}^{occ} C_{pi} C_{iq}$ 

$C_{pq}$ being the transformed (non-orthogonal) orbital eigenvectors, $F_{pq}$ the Fock matrix $=H_{pq}^{core} + \displaystyle \sum_{rs} D_{pq} \[2(pq \vert rs) - (pr \vert qs) \]$ and $H_{pq}^{core}$ is the core Hamiltonian-the sum of 1-electron Coulomb and kinetic energy integrals. To this you add the nuclear repulsion energy. A good explanation can be found in [Crawford Programming Projects #3](https://github.com/CrawfordGroup/ProgrammingProjects/tree/master/Project%2303). I hope this helps.

**Comment by sunqm** [comment]
I'll close this issue as the answers should be clear enough for the energy formula
